### ISSUE TITLE
become_dask tries to connect to the wrong IP

### ISSUE THREAD

**Issue by sk1p** [ipython/ipyparallel#414]
When trying to use `Client.become_dask`, the dask workers try to connect to the scheduler at `0.0.0.0` instead of the external IP address. This appears to be the culprit:

https://github.com/ipython/ipyparallel/blob/master/ipyparallel/controller/hub.py#L1477-L1481

This is how it looks like on my system:

```
In [3]: from distributed import Scheduler                                       
In [4]: scheduler = Scheduler({})                          
                     
In [7]: await scheduler.start()                                                 
Out[7]: <Scheduler: "tcp://192.168.178.27:46229" processes: 0 cores: 0>

In [9]: scheduler.ip                                                            
Out[9]: '0.0.0.0'

In [11]: scheduler.address                                                      
Out[11]: 'tcp://192.168.178.27:46229'
```

The error happened with `distributed` 2.8.1 and ipyparallel 6.3.0 (at least on the client side, it is unclear to me if the controller version matched); the ipython session above was using a more current distributed 2.17.0

I don't have the exact traceback handy right now, maybe @jhgee can add it here as a comment.

Thanks!

/cc @uellue

**Comment by jhgee** [comment]
```---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
~/.local/lib/python3.6/site-packages/distributed/comm/core.py in connect(addr, timeout, deserialize, connection_args)
    221             if not comm:
--> 222                 _raise(error)
    223         except FatalCommClosedError:

~/.local/lib/python3.6/site-packages/distributed/comm/core.py in _raise(error)
    204         )
--> 205         raise IOError(msg)
    206 

OSError: Timed out trying to connect to 'tcp://0.0.0.0:33557' after 10 s: in <distributed.comm.tcp.TCPConnector object at 0x2aac05c509b0>: ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

OSError                                   Traceback (most recent call last)
<ipython-input-10-312fab997008> in <module>
      6 }
      7 
----> 8 client = c.become_distributed(scheduler_args=scheduler_args)
      9 ctx = api.Context(executor=DaskJobExecutor(client=client))

~/.local/lib/python3.6/site-packages/ipyparallel/client/client.py in become_dask(self, targets, port, nanny, scheduler_args, **worker_args)
   1383             distributed_Client = distributed.Executor
   1384 
-> 1385         client = distributed_Client('{ip}:{port}'.format(**distributed_info))
   1386 
   1387         return client

~/.local/lib/python3.6/site-packages/distributed/client.py in __init__(self, address, loop, timeout, set_as_default, scheduler_file, security, asynchronous, name, heartbeat_interval, serializers, deserializers, extensions, direct_to_workers, **kwargs)
    724             ext(self)
    725 
--> 726         self.start(timeout=timeout)
    727         Client._instances.add(self)
    728 

~/.local/lib/python3.6/site-packages/distributed/client.py in start(self, **kwargs)
    889             self._started = asyncio.ensure_future(self._start(**kwargs))
    890         else:
--> 891             sync(self.loop, self._start, **kwargs)
    892 
    893     def __await__(self):

~/.local/lib/python3.6/site-packages/distributed/utils.py in sync(loop, func, callback_timeout, *args, **kwargs)
    332     if error[0]:
    333         typ, exc, tb = error[0]
--> 334         raise exc.with_traceback(tb)
    335     else:
    336         return result[0]

~/.local/lib/python3.6/site-packages/distributed/utils.py in f()
    316             if callback_timeout is not None:
    317                 future = gen.with_timeout(timedelta(seconds=callback_timeout), future)
--> 318             result[0] = yield future
    319         except Exception as exc:
    320             error[0] = sys.exc_info()

~/.local/lib/python3.6/site-packages/tornado/gen.py in run(self)
    733 
    734                     try:
--> 735                         value = future.result()
    736                     except Exception:
    737                         exc_info = sys.exc_info()

~/.local/lib/python3.6/site-packages/distributed/client.py in _start(self, timeout, **kwargs)
    982 
    983         try:
--> 984             await self._ensure_connected(timeout=timeout)
    985         except OSError:
    986             await self._close()

~/.local/lib/python3.6/site-packages/distributed/client.py in _ensure_connected(self, timeout)
   1039                 self.scheduler.address,
   1040                 timeout=timeout,
-> 1041                 connection_args=self.connection_args,
   1042             )
   1043             comm.name = "Client->Scheduler"

~/.local/lib/python3.6/site-packages/distributed/comm/core.py in connect(addr, timeout, deserialize, connection_args)
    229                 logger.debug("sleeping on connect")
    230             else:
--> 231                 _raise(error)
    232         else:
    233             break

~/.local/lib/python3.6/site-packages/distributed/comm/core.py in _raise(error)
    203             error,
    204         )
--> 205         raise IOError(msg)
    206 
    207     # This starts a thread

OSError: Timed out trying to connect to 'tcp://0.0.0.0:33557' after 10 s: Timed out trying to connect to 'tcp://0.0.0.0:33557' after 10 s: in <distributed.comm.tcp.TCPConnector object at 0x2aac05c509b0>: ConnectionRefusedError: [Errno 111] Connection refused
```

**Comment by sk1p** [comment]
@jhgee I tried to fix this in #415, could you see if the fix works in the context of your slurm cluster? Thanks!
